WIP: Still fleshing out some code samples and framing [draft]

nathan-contino opened the pull request:
Not ready for review just yet. Just posting in case I get hit by a bus.

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+| `spi_bus` | integer | Optional | SPI bus number used to connect the gateway peripheral. <br> Options: `0`, `1` |
+| `region_code` | string | Optional | Frequency region of your gateway. <br> Options: `US915`, `EU868` <br> Default: `US915` |
+| `reset_pin` | integer | **Required** | GPIO pin used for SX1302 reset. <br> Not configurable for `sx1302-waveshare-hat`. |
+| `power_en_pin` | integer | Optional | GPIO pin used for SX1302 power enable. <br> Not configurable for `sx1302-waveshare-hat`. |
```
just realized its not documented in the readme yet but the generic hat also has an optional `path` attribute for the serial path, used if the sx1302 connects through USB

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+| ---- | ---- | --------- | ----------- |
+| `board` | string | **Required** | Name of the [board component](/operate/reference/components/board/) that represents the Raspberry Pi Compute Module inside the RAK7391. Used for GPIO pin control. |
+| `region_code` | string | Optional | Frequency region of your gateway. <br> Options: `US915`, `EU868` <br> Default: `US915` |
+| `pcie1` | object | optional | PCIe configuration for a concentrator: <br> <ul><li>`spi_bus` (integer) (Optional): Serial bus that the concentrator is connected to, if connected through SPI. </li><li>`serial_path` (string) (Optional): Serial path that the concentrator is mounted at, if connected through USB. </li></ul> Not configurable for `sx1302-waveshare-hat`. |
```
```suggestion
| `pcie1` | object | optional | PCIe configuration for a concentrator: <br> <ul><li>`spi_bus` (integer) (Optional): SPI bus that the concentrator is connected to, if connected through SPI. </li><li>`serial_path` (string) (Optional): Serial path that the concentrator is mounted at, if connected through USB. </li></ul> Not configurable for `sx1302-waveshare-hat`. |
```

oliviamiller replied:
the reason these are called pcie1 and pcie2 is because that's how the slots are labeled on the rak hardware, might want to call out that you cannot interchange pcie1 and pcie2

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+| `board` | string | **Required** | Name of the [board component](/operate/reference/components/board/) that represents the Raspberry Pi Compute Module inside the RAK7391. Used for GPIO pin control. |
+| `region_code` | string | Optional | Frequency region of your gateway. <br> Options: `US915`, `EU868` <br> Default: `US915` |
+| `pcie1` | object | optional | PCIe configuration for a concentrator: <br> <ul><li>`spi_bus` (integer) (Optional): Serial bus that the concentrator is connected to, if connected through SPI. </li><li>`serial_path` (string) (Optional): Serial path that the concentrator is mounted at, if connected through USB. </li></ul> Not configurable for `sx1302-waveshare-hat`. |
+| `pcie2` | object | Optional | PCIe configuration for a second concentrator: <br> <ul><li>`spi_bus` (integer) (Optional): Serial bus that the concentrator is connected to, if connected through SPI. </li><li>`serial_path` (string) (Optional): Serial path that the concentrator is mounted at, if connected through USB. </li></ul> Not configurable for `sx1302-waveshare-hat`. |
```
the section is titled configuration for RAK7391 gateways so I think you can remove the stuff about the waveshare hat

nathan-contino replied:
```suggestion
| `pcie2` | object | Optional | PCIe configuration for concentrator connected to PCIe slot 2: <br> <ul><li>`spi_bus` (integer) (Optional): Serial bus that the concentrator is connected to, if connected through SPI. </li><li>`serial_path` (string) (Optional): Serial path that the concentrator is mounted at, if connected through USB. </li></ul> |
```

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+{{% /tab %}}
+{{% tab name="JSON Configuration" %}}
+
+In the `components` section of your machine configuration, add the following object, depending on your preferred [activation protocol](#activation-protocols):
```
might be helpful to call out that most nodes are OTAA by default

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+    "app_s_key": <application-session-key>,
+    "network_s_key": <network-session-key>,
+    "gateways": [<gateway-name>]
+  }
```
the milesight supported devices have a default app key and session keys so the user likely wouldn't have to supply them, just something to note

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+- `viam:lorawan:node`: Any LoRaWAN sensor that is:
+  - Class A
+  - supports either the `US915` or `EU868` frequency band
+  - uses LoRaWAN MAC specification version 1.0.3 or later
```
The module is not compatible with version 1.1, I think it would be compatible with any 1.0.x but only 1.0.3 has been tested

nathan-contino replied:
```suggestion
  - uses LoRaWAN MAC specification version 1.0.3
```
Sounds like we're better off sticking with the specific version, thanks!

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+| `app_s_key` | string | Optional | 128-bit hexadecimal **application session key** used to decrypt uplink messages. Required for ABP activation protocol. Found in the device datasheet or in device packaging. <br> On `milesight-ct101` and `milesight-em310-tilt`, defaults to `5572404C696E6B4C6F52613230313823`. |
+| `network_s_key` | string | Optional | 128-bit hexadecimal **network session key** used to decrypt uplink messages. Required for ABP activation protocol. Found in the device datasheet or in device packaging. <br> On `milesight-ct101` and `milesight-em310-tilt`, defaults to `5572404C696E6B4C6F52613230313823`. |
+| `gateways` | string[] | **Required** | Name of the [gateway component](#add-a-gateway) in your Viam configuration. |
+| `uplink_interval_mins` | decimal | **Required** | Interval between uplink messages sent from the node, in minutes. Found in the device datasheet, but can be modified. <br> On `milesight-ct101`, defaults to `10`. On `milesight-em310-tilt`, defaults to `1080`. |
```
might be helpful to call out that the module will modify the uplink to the one provided in the config using a downlink for the non-generic nodes

nathan-contino replied:
```suggestion
| `uplink_interval_mins` | decimal | **Required** | Interval between uplink messages sent from the node, in minutes. Found in the device datasheet, but can be modified. Configured by downlink after initial connection. <br> On `milesight-ct101`, defaults to `10`. On `milesight-em310-tilt`, defaults to `1080`. |
```

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+| Activation By Personalization | `ABP` | Static | Static unless manually rotated | <ul><li> `dev_addr` </li><li> `network_s_key` </li><li> `app_s_key` </li></ul> |
+
+To specify an activation protocol for your network, use the `join_type` field.
+You must specify this field for both gateways and nodes.
```
should be just nodes, did you see something that made you think differently?

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
````diff
@@ -0,0 +1,513 @@
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+```python
````
I would expect that the calibration commands would be easiest to use from within the app UI (since the sensor needs to be in specific solutions before sending the downlinks). Might just want to include that its also an options to do these commands from the user interface

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+  - Check the power, transmit, and receive LEDs for activity. Most SX1302 gateways have three LEDs:
+    - **Power LED (red, solid)**: gateway receiving power
+    - **TX LED (red, blinking)**: gateway transmitting data
+    - **RX LED (red, blinking)**: gateway receiving data
```
these leds are just for the waveshare HAT

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+
+  - Check that `dev_eui` in your node configuration matches the value of your device.
+  - If you use the OTAA activation protocol, confirm that `app_key` matches the application key defined in the device datasheet.
+  - If you use the ABP activation protocol, verify that `app_s_key` and `network_s_key` match the values defined in the device datasheet or device packaging.
```
might want to include that restarting the node can help to trigger a join request, and that it may take up to 10-15 minutes for the device to join the network. I've seen lots of people think it doesn't work when it doesn't join right away.

oliviamiller commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+To start your network, you need a gateway.
+The lorawan module supports the following varieties of gateway hardware:
+
+- peripherals built on the SX1302 chip such as the [Waveshare SX1302 Gateway HAT](https://www.waveshare.com/wiki/SX1302_LoRaWAN_Gateway_HAT), which you can connect to an SBC
```
so the model names are unfortunate but we have also now tested the sx1303 so may want to include somewhere that is also supported!

oliviamiller replied:
yes!

oliviamiller reviewed (commented):
looks really good!! left some suggestions but feel free to ignore any you don't agree with

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
````diff
@@ -0,0 +1,513 @@
+1. Enable SPI on your machine:
+
+   ```console
+   sudo raspi-config nonint do_spi 0
````
mostly a CYA, note that I also call it out in the troubleshooting steps :)

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+
+If you choose the RAK7391:
+
+1. Complete the [quickstart](https://docs.rakwireless.com/product-categories/wisgate/rak7391/quickstart/) to flash your RAK7391 with an operating system and connect it to the network.
```
oooh thank you! I think they might have a few too many things called quickstarts, because I found the one I linked and left it there. But yours is much more straightforward since it's purely OS oriented.

```suggestion
1. Complete the [RAKPiOS quickstart](https://docs.rakwireless.com/product-categories/software-apis-and-libraries/rakpios/quickstart/) to flash your RAK7391 with an operating system and connect it to the network.
```

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+
+  - If using a HAT, ensure your HAT is properly seated on the SBC's GPIO header.
+  - Check that the antenna is securely connected to your gateway.
+  - Check the power, transmit, and receive LEDs for activity. Most SX1302 gateways have three LEDs:
```
```suggestion
  - On the Waveshare SX1302 LoRaWAN Gateway HAT, check the status LEDs for activity:
```

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,513 @@
+  - Check that `spi_bus` is configured for the correct SPI bus.
+
+### Sensor fails to join network
+
```
```suggestion

- **Wait 10-15 minutes**:
  - Nodes can take up to 10-15 minutes to join the network.
  - If a node doesn't join the network within 30 minutes, try restarting the node.
```

nathan-contino commented:
@JessamyT I discussed some options for my approach with Naomi and some other stakeholders a week or two ago. I believe we specifically discussed the [cartographer service page](https://docs.viam.com/operate/reference/services/slam/cartographer/), the [mlmodel service page](https://docs.viam.com/operate/reference/services/vision/mlmodel/), and [alert on machine telemetry](https://docs.viam.com/manage/troubleshoot/alert/).

I'm increasingly leaning towards pulling this out of reference, though, because there is a fair amount of conceptual and how-to content that you inevitably have to cover before you can talk about LoRaWAN.

Mostly because our reference pages are... difficult to find, to say the least. They're almost a secret!

Any suggestions for where we could put this how-to? Maybe under [Connect devices](https://docs.viam.com/operate/)? But it doesn't feel as introductory as most of that material. I'd appreciate your thoughts!

JessamyT commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+Use the `fport` field of a node configuration to specify the frame port value that the node expects to use for downlink communication.
+When you specify an `fport` value for a node, gateways use that value as the frame port for all downlink communication with the node.
+
+## Control nodes directly
```
Naively, I find myself wondering "wait, did I just miss something about controlling them indirectly?" I also find myself wondering "Are there do commands to control the gateway?" Maybe add a sentence transitioning into this section?

Otherwise this seems like a very nice user-friendly section :D

nathan-contino replied:
I don't think there are any gateway docommands, but I will double check that!

JessamyT replied:
If not, [opt] add something to this transition that says so, along with the direct vs indirect bit

JessamyT commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+
+## Hardware requirements
+
+- a [supported LoRaWAN gateway](https://github.com/viam-modules/lorawan?tab=readme-ov-file#lorawan-gateway-models-provided)
+- a [supported LoRaWAN node](https://github.com/viam-modules/lorawan?tab=readme-ov-file#lorawan-sensor-models-provided)
```
```suggestion
- A [supported LoRaWAN gateway](https://github.com/viam-modules/lorawan?tab=readme-ov-file#lorawan-gateway-models-provided)
- A [supported LoRaWAN node](https://github.com/viam-modules/lorawan?tab=readme-ov-file#lorawan-sensor-models-provided)
```
[nit]

JessamyT commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+
+## Architecture
+
+You can use Viam to create LoRaWAN networks containing one gateway and multiple nodes.
```
```suggestion
You can create LoRaWAN networks containing one gateway and multiple nodes.
```
Since we're already in the Viam docs we don't have to say this, but can just focus on their end goal

JessamyT commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+- **uplink** transmits from nodes to gateways
+- **downlink** transmits from gateways to nodes
+
+![A LoRaWAN network consisting of a single gateway and multiple nodes](/components/sensor/lorawan.png)
```
[opt] Gets a smidge big on very wide screens; consider setting the width and adding the imgzoom class
But maybe you're getting a pretty visual from Cristina anyway?

nathan-contino replied:
That's the plan. Let me know if you have any suggestions for the visual before I send in my request; I'd rather not bother Cristina more than once!

JessamyT commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+
+![A LoRaWAN network consisting of a single gateway and multiple nodes](/components/sensor/lorawan.png)
+
+### Node
```
```suggestion
### Nodes
```
[opt] Could be either, but this might emphasize that this one can be multiple?

Also consider putting gateway then node in the same order here as in configuration below

JessamyT commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+| `network_s_key` | string | Optional | 128-bit hexadecimal **network session key** used to decrypt uplink messages. Required for ABP activation protocol. Found in the device datasheet or in device packaging. <br> On `milesight-ct101` and `milesight-em310-tilt`, defaults to `5572404C696E6B4C6F52613230313823`. |
+| `gateways` | string[] | **Required** | Name of the [gateway component](#add-a-gateway) in your Viam configuration. |
+| `uplink_interval_mins` | decimal | **Required** | Interval between uplink messages sent from the node, in minutes. Found in the device datasheet, but can be modified. Configured by downlink after initial connection. <br> On `milesight-ct101`, defaults to `10`. On `milesight-em310-tilt`, defaults to `1080`. |
+| `decoder_path` | string | **Required** | Path to a Javascript **decoder script** used to interpret data transmitted from the node. You can use a local path on your device or an HTTP(S) URL that points to a file on a remote server. If the decoder script provides multiple implementations, uses the Chirpstack version. Not compatible with The Things Network decoders. <br> On supported Milesight and ...
```
For all the tables, if they do stay in instead of going into readmes, maybe move all the required attributes to the top, and/or make the tables match what's in the readme (if possible?) so that if I have the readme open and this open I don't wonder "hmm are all the same attributes listed in both places or not? I'm cross-checking and kinda overwhelmed!"

nathan-contino replied:
Trouble is, different modules have different required attributes, and depending on your configuration, certain attributes are required or optional. I've tried to put these in order from most->least likely to be relevant, unfortunately a lot of them are relevant!

JessamyT commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+<!-- prettier-ignore -->
+| Name | Value | Key | Session Key | Configuration |
+| ---- | ----- | --- | ----------- | ------------- |
+| Over-The-Air Activation | `OTAA` | Dynamic, generated at join time | Automatically rotated | <ul><li> `dev_eui` </li><li> `app_eui` </li><li> `app_key` </li></ul> |
```
```suggestion
| Over-the-air activation | `OTAA` | Dynamic, generated at join time | Automatically rotated | <ul><li> `dev_eui` </li><li> `app_eui` </li><li> `app_key` </li></ul> |
```
[nit]

nathan-contino replied:
I'm going to ignore this suggestion because this is an acronym and I'm using the [official capitalization](https://www.thethingsnetwork.org/docs/lorawan/end-device-activation/)!

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+To start your network, you need a gateway.
+The lorawan module supports the following varieties of gateway hardware:
+
+- SBC peripherals built on the SX1302 or SX1303 chips, such as the [Waveshare SX1302 Gateway HAT](https://www.waveshare.com/wiki/SX1302_LoRaWAN_Gateway_HAT)
```
Great point. I tweaked this wording from 'peripheral connected to an SBC' and it seems like I overcorrected!

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+
+If you choose the RAK7391:
+
+1. Complete the [RAKPiOS quickstart](https://docs.rakwireless.com/product-categories/software-apis-and-libraries/rakpios/quickstart/) to flash your RAK7391 with an operating system and connect it to the network.
```
```suggestion
1. Complete the [RAKPiOS quickstart](https://docs.rakwireless.com/product-categories/software-apis-and-libraries/rakpios/quickstart/) to flash your RAK7391 with an operating system and connect it to the Internet.
```

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+If you choose the RAK7391:
+
+1. Complete the [RAKPiOS quickstart](https://docs.rakwireless.com/product-categories/software-apis-and-libraries/rakpios/quickstart/) to flash your RAK7391 with an operating system and connect it to the network.
+1. [Install viam-server](https://docs.viam.com/operate/get-started/setup/).
```
LOL. That explains how I kept winding up in the Real Docs during testing. Thanks!

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+   If your SX1302- or SX1303-based peripheral lacks a dedicated model, choose `lorawan:sx1302-hat-generic`.
+1. Click **Add module**, and enter a name for your gateway.
+1. Click **Create** to add the module to your machine.
+1. Click **Save** in the top right to apply your changes and load your new module.
```
```suggestion
1. Click **Save** in the top right to save your configuration.
```

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
````diff
@@ -0,0 +1,519 @@
+}
+```
+
+Choose an appropriate board model based from the [board components registry](/operate/reference/components/board/).
````
```suggestion
Choose an appropriate board model from the [board components registry](/operate/reference/components/board/).
```
Thanks! Missed this when applying eng suggestions.

nathan-contino commented:
Un-hiding components in this PR because otherwise you can't navigate to this page without the link.

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+
+Next, add a `lorawan` gateway component:
+
+1. Click the **+** icon next to your machine part in the left-hand menu and select **Component or service**.
```
I haven't seen this one yet in production, so I'm holding off for now. I assume we should bold **add another component** since it's a UI element? And possibly capitalize Add?

JessamyT replied:
Since there's not a UI element literally called "add another component" no, you wouldn't bold it. I'm just suggesting words and letting the user use just a smidge of inference since you've already walked them through it, and I actually do similar things in places... not sure exactly where off hand

JessamyT replied:
Ooh found one: https://docs.viam.com/operate/mobility/move-base/#configure-and-connect-to-your-base
See also move a gantry, and https://docs.viam.com/manage/software/deploy-software/#configure-your-hardware-and-software

nathan-contino commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+{{% /tab %}}
+{{< /tabs >}}
+
+The following attributes are available for SX1302-based LoRaWAN gateways:
```
I'll definitely PR these improved wording updates into the README. I _think_ the information is still useful here -- I certainly would want it when I'm learning about the concept! -- but I could be convinced otherwise.

JessamyT commented:
> Un-hiding components in this PR because otherwise you can't navigate to this page without the link.

Technically you can--you just have to go component APIs -> sensor API -> click the link in the first sentence -> model page 😈 Easy peasy XD

nathan-contino commented:
Truly a set of instructions worth of the smiling imp emoji!

JessamyT commented on docs/operate/reference/components/sensor/lorawan.md:
```diff
@@ -0,0 +1,519 @@
+{{< tabs >}}
+{{% tab name="Builder" %}}
+
+1. Navigate to the **CONFIGURE** tab of your machine's page in the [Viam app](https://app.viam.com).
+1. Click the **+** icon next to your machine part in the left-hand menu and select **Component or service**.
```
You have already have them navigate to the config tab though. We've gotten feedback in the past that we are overly-hand-holdy in some more obvious steps, and then throw the user to the wind later. I wish I could remember where I've done this, but here's one example: https://docs.viam.com/manage/software/deploy-software/#configure-your-hardware-and-software In any case if it makes sense to you to do, can do it!

npentrel commented on docs/data-ai/capture-data/lorawan.md:
```diff
@@ -0,0 +1,522 @@
+[LoRaWAN (Long Range Wide Area Network)](https://lora-alliance.org/) enables sensor communication spanning kilometers with minimal power usage.
+
+To collect LoRaWAN sensor data with Viam, use the [`lorawan`](https://app.viam.com/module/viam/lorawan) {{< glossary_tooltip term_id="module" text="module" >}}.
+You can find the module's source code in the [lorawan GitHub repository](https://github.com/viam-modules/lorawan).
```
```suggestion
```

nathan-contino replied:
Removed!

npentrel commented on docs/data-ai/capture-data/lorawan.md:
```diff
@@ -0,0 +1,522 @@
+### Gateway
+
+Gateways collect data generated by LoRaWAN nodes.
+A gateway is a device that communicates with nodes over LoRaWAN.
```
```suggestion
The gateway is the device that communicates with nodes over LoRaWAN and that connects to Viam.
```

nathan-contino replied:
Incorporating this feedback in my next commit, thanks.

npentrel commented on docs/data-ai/capture-data/lorawan.md:
```diff
@@ -0,0 +1,522 @@
+| `app_key` | string | Optional | The 128-bit hexadecimal AES **application key** used for device authentication and session key derivation. Required for OTAA activation protocol. Found in the device datasheet. <br> On `milesight-ct101` and `milesight-em310-tilt`, defaults to `5572404C696E6B4C6F52613230313823`. |
+| `gateways` | string[] | **Required** | Name of the [gateway component](#add-a-gateway) in your Viam configuration. |
+
+For generic nodes, you must also configure a [`decoder_path`](#decoder-script).
```
what are generic nodes? 

Also the connection between decoder_path and what it should link to is a bit missing:
```suggestion
For generic nodes, you must also configure a `decoder_path` for the [decoder script](#decoder-script).
```

npentrel reviewed (commented):
Some minor comments. Mostly I think I wonder if frame port and decoder script really need to have their own sections rather than being explained in the instructions or in the attribute information

nathan-contino commented on docs/data-ai/capture-data/lorawan.md:
```diff
@@ -0,0 +1,522 @@
+Gateways collect data generated by LoRaWAN nodes.
+A gateway is a device that communicates with nodes over LoRaWAN.
+Add a gateway to a Viam machine to sync data from many sensors to a single collection in the Viam cloud, where you can aggregate and visualize your data.
+With Viam's `lorawan` module, your gateway also acts as a network server.
```
Agreed, removing this sentence in my next commit.

nathan-contino commented on docs/data-ai/capture-data/lorawan.md:
```diff
@@ -0,0 +1,522 @@
+
+You must configure the following attributes for SX1302 and SX1303-based LoRaWAN gateways:
+
+<!-- prettier-ignore -->
```
I eliminated everything but the essentials, but I think this comment convinces me to transmute these into bullet points instead. Tables are a tad too formal/referential for this context.

nathan-contino commented:
@npentrel @JessamyT when you get the chance (i know this is a busy week with other things), give this another look and let me know if you see any other issues. I should have addressed all of your feedback. I'd be happy to merge this sooner and follow-up with a PR to add the updated diagram next week, but I'll defer to your stance on diagrams.

npentrel commented on docs/data-ai/capture-data/lorawan.md:
```diff
@@ -333,8 +337,25 @@ await node.doCommand({ restart_sensor: {} });
+{{< tabs >}}
+{{% tab name="Viam web app" %}}
+
+To restart a node from the Viam web app, specify a hexadecimal string in the following DoCommand input in the **CONTROL** tab of your machine page:
```
```suggestion
To restart a node from the Viam web UI, specify a hexadecimal string in the following DoCommand input in the **CONTROL** tab of your machine page:
```

npentrel commented on docs/data-ai/capture-data/lorawan.md:
```diff
@@ -461,19 +503,138 @@ await node.doCommand({'calibrate_orp': {}});
+
+Use the calibration processes below to calibrate your sensor with these commands:
+
+#### Calibrate the pH probe
```
Is this a common enough thing that we want to cover it here? Leaving that up to you but it feels odd to me

nathan-contino replied:
It is a requirement to properly use this sensor model, which is one of our small number of officially supported sensors. And the pattern is generalizable across all other sensors that require calibration. So I'm leaning 'include' unless someone feels really strongly otherwise. I understand how it might feel odd but I think it's important to draw attention to this so folks don't wind up with a lot of uncalibrated sensors with unreliable readings.

nathan-contino commented on docs/data-ai/capture-data/lorawan.md:
```diff
@@ -0,0 +1,683 @@
+  - supports either the `US915` or `EU868` frequency band
+  - uses LoRaWAN MAC specification version 1.0.3
+
+Configure attributes based on the tables below:
```
```suggestion
Configure attributes based on the descriptions below:
```

JessamyT commented on docs/data-ai/capture-data/lorawan.md:
```diff
@@ -0,0 +1,686 @@
+   If the name of your node does not appear in the list, choose the generic `lorawan:node` option.
+1. Click **Add module**, and enter a name for your node.
+1. Click **Create** to add the model and module to your machine.
+1. Configure the model attributes as follows, according to the node type:
+1. Click **Save** in the top right to apply your changes.
```
Sorry I may have made a mistake here in my suggestion in the last review.
```suggestion
1. Configure the model attributes as follows, according to the node type:
```
or something like this, or move "Click save..." to the end
```suggestion
1. Configure the model attributes as described below, according to the node type.
1. Click **Save** in the top right to apply your changes.
```

nathan-contino commented:
Merging now, will update when the improved image drops!
